examples/terraform-aws-s3-example/outputs.tf

```
@@ -6,3 +6,6 @@ output "bucket_arn" {
   value = aws_s3_bucket.test_bucket.arn
 }
 
+output "logging_target_bucket" {
+  value = aws_s3_bucket.test_bucket.logging[*].target_bucket
```

NIT: Use `[0]` so that we return a string instead of a list:
```suggestion
  value = aws_s3_bucket.test_bucket.logging[0].target_bucket
```

---

This is what i tried initially, but then came across [this issue:](https://github.com/hashicorp/terraform/issues/22504)
```
Block type "logging" is represented by a set of objects, and set elements do
        	            	not have addressable keys. To find elements matching specific criteria, use a
        	            	"for" expression with an "if" clause.
```

I thought the simplest way is leave this as is. We have a few other ways we could go about this to make it clearer though:

- add a comment above this line
- use `tolist(aws_s3_bucket.test_bucket.logging)[0].target_bucket` (I've tested this and it seemed to work well!)
- or try changing the lifecycle policy for this: _(as suggested in some of the issue threads)_
```
lifecycle {
     create_before_destroy = true
     ignore_changes = [
       # Ignore changes to first tag in list: "foo"
       tags[0]
     ]
   }
```

I'd probably go with the 2nd option - use the `tolist` tf function, so it's clear to users of this example this returns a list

---

Ah gotcha. Yes the `tolist` approach works for me!